### ISSUE TITLE
There is not any projects in xcworkspace if recreate the xcworkspace 

### ISSUE THREAD

**Issue by Whirlwind** [CocoaPods/CocoaPods#9237]
After `pod install`, everything is OK. If I delete the xcworkspace bundler, then rerun the `pod install`, I will get a new xcworkspace bundler. But there is not any user projects in the xcworkspace, only a `Pods` project.

I check the code in `user_project_integrator.rb`, I find that:
```
      def user_project_paths_to_integrate
        targets_to_integrate.map(&:user_project_path).compact.uniq
      end
      def create_workspace
        all_projects = user_project_paths_to_integrate.sort.push(sandbox.project_path).uniq
        file_references = all_projects.map do |path|
          relative_path = path.relative_path_from(workspace_path.dirname).to_s
          Xcodeproj::Workspace::FileReference.new(relative_path, 'group')
        end

        if workspace_path.exist?
          workspace = Xcodeproj::Workspace.new_from_xcworkspace(workspace_path)
          new_file_references = file_references - workspace.file_references
          unless new_file_references.empty?
            new_file_references.each { |fr| workspace << fr }
            workspace.save_as(workspace_path)
          end

        else
          UI.notice "Please close any current Xcode sessions and use `#{workspace_path.basename}` for this project from now on."
          workspace = Xcodeproj::Workspace.new(*file_references)
          workspace.save_as(workspace_path)
        end
      end
```

Because I have the all cached pod projects, include the Pods.xcodeproj. So the `targets_to_integrate` is empty, it will get an empty array from the `user_project_paths_to_integrate`, and will not add user project into the xcworkspace.

It seems that it is a small bug.

I am using CocoaPods v1.7.

**Comment by dnkoutso** [comment]
Yeap probably a bug in incremental installation, do you have that enabled just to verify?

If so we might want to add the workspace as part of the project cache? cc @sebastianv1

**Comment by Whirlwind** [comment]
> Yeap probably a bug in incremental installation, do you have that enabled just to verify?

Yes, I enable the  incremental installation. And hit all caches.

**Comment by sebastianv1** [comment]
PR: #9241

**Comment by dnkoutso** [comment]
fixed for 1.9.x
